Hw 13 1 create annotation

NikolayAbysov opened the pull request.

NikolayAbysov committed:
Solution implemented

NikolayAbysov committed:
Solution implemented

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,32 @@
+
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getName().equalsIgnoreCase("betDao")
```
```suggestion
                    && field.getType().equals(BetDao.class)
```
краще тут не прив'язуватись до неймінгу поля а до його типу

NikolayAbysov replied:
Fixed that, thanks :)

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,32 @@
+                field.setAccessible(true);
+                field.set(instance, Factory.getBetDao());
+            } else if (field.getAnnotation(Inject.class) != null
+                    && field.getName().equalsIgnoreCase("playerDao")) {
```
аналогічно тут, і з PlayerDao теж потрібно перевіряти чи над класом стоїться аннотація @Dao

NikolayAbysov replied:
Fixed that

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,32 @@
+                field.set(instance, Factory.getPlayerDao());
+            }
+        }
+        return instance;
```
а що буде якщо @Dao не буде стояти над класом? давай будемо кидати ексепшин, який самі створимо?

RomanDubovskyi commented on src/main/java/core/basesyntax/dao/BetDaoJImpl.java:
```diff
@@ -0,0 +1,15 @@
+
+import core.basesyntax.model.Bet;
+import java.util.List;
+
```
давай тут теж додамо аннотація Dao

NikolayAbysov committed:
Issues fixed:

1 checks now mapped to field types not names
2 new custom exception for @Dao annotation implemented

RomanDubovskyi commented on src/main/java/core/basesyntax/dao/BetDaoJImpl.java:
```diff
@@ -0,0 +1,15 @@
+import core.basesyntax.model.Bet;
+import java.util.List;
+
+public class BetDaoJImpl implements BetDao {
```
я так розумію цей клас який скопійований з відео Богдана і називався там BetDaoJDBCImpl
він нам тут не потрібний, давай його видалимо

RomanDubovskyi reviewed (approved):
Хороша робота, тільки не забудь видалити непотрібний клас з відео
